File: challenges/8-contador-nucleotideos/java/maikella/ContadorDeNucleotideos.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class ContadorDeNucleotideos {

   public static void main(String[] args) {

      dna();
   }


   private static void dna() {

      ArrayList<String> dna = new ArrayList<>();
      Scanner s = new Scanner(System.in);

      int a = 0;
      int c = 0;
      int g = 0;
      int t = 0;

      System.out.println("Entre com uma sequência de DNA: ");
      String dnaSequence = s.next().toUpperCase();

      if(!(dnaSequence.contains("A") && dnaSequence.contains("C") && dnaSequence.contains("G") && dnaSequence.contains("T"))){
         System.out.println("Sequência de DNA inválida");
      }else{

         for (int i = 0; i < dnaSequence.length(); i++) {

            dna.add(String.valueOf(dnaSequence.charAt(i)));

            if(dna.get(i).contains("A")){
               a++;
            }if(dna.get(i).contains("C")){
               c++;
            }if(dna.get(i).contains("G")){
               g++;
            }if(dna.get(i).contains("T")){
               t++;
            }

         }

         System.out.println(String.format("'A' : %s, 'C' : %s, 'G' : %s, 'T' : %s", a, c, g, t));
      }
   }

}

File: challenges/8-contador-nucleotideos/typescript/marcosvag/contadorDeNucleotideos.ts
const input = process.argv[2];

function contadorDeNucleotideos(sequenciaDeDNA: string): void {
    const nucleotideos: { [nucleotideo: string]: number } = {
        "A": 0,
        "C": 0,
        "G": 0,
        "T": 0,
    }

    for (let e = 0; e < sequenciaDeDNA.length; e++) {
        if (sequenciaDeDNA[e] in nucleotideos) {
            nucleotideos[sequenciaDeDNA[e]]++;
        } else {
            return console.error(`"Sequência de DNA inválida"`);
        }
    }
    return console.log(`"'A': ${nucleotideos["A"]}, 'C': ${nucleotideos["C"]},'G': ${nucleotideos["G"]},'T': ${nucleotideos["T"]}"`);
}

contadorDeNucleotideos(input);